src/binary.js

```diff
@@ -155,46 +187,57 @@ class Writer {
 
         this.writeAdaptiveNumber(strBuffer.length << 1);
 
+        this.ensureCapacity(strBuffer.length);
```

Comment:
There will be an error here if the `strBuffer` size is larger than the `chunkSize`. A test for this case is missed.
You need to add a set of subarrays of `strBuffer` into `chunks`.